armi/utils/tests/test_tabulate.py

```diff
@@ -0,0 +1,1697 @@
+# Copyright 2024 TerraPower, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for tabulate.
+
+This file started out as the MIT-licensed "tabulate". Though we have made, and will continue to make
+many arbitrary changes as we need. Thanks to the tabulate team.
+
+https://github.com/astanin/python-tabulate
+"""
+from collections import namedtuple
+from collections import OrderedDict
+from collections import UserDict
+from dataclasses import dataclass
+from datetime import datetime
+import unittest
+
+import numpy as np
+
+from armi.utils.tabulate import _alignCellVeritically
+from armi.utils.tabulate import _alignColumn
+from armi.utils.tabulate import _bool
+from armi.utils.tabulate import _buildLine
+from armi.utils.tabulate import _buildRow
+from armi.utils.tabulate import _format
+from armi.utils.tabulate import _isMultiline
+from armi.utils.tabulate import _multilineWidth
+from armi.utils.tabulate import _normalizeTabularData
+from armi.utils.tabulate import _table_formats
+from armi.utils.tabulate import _type
+from armi.utils.tabulate import _visibleWidth
+from armi.utils.tabulate import _wrapTextToColWidths
+from armi.utils.tabulate import SEPARATING_LINE
+from armi.utils.tabulate import tabulate
+from armi.utils.tabulate import tabulate_formats
+
+
+class TestTabulateAPI(unittest.TestCase):
+    def test_tabulateFormats(self):
+        """API: tabulate_formats is a list of strings."""
+        supported = tabulate_formats
```

Comment:
🐍 case -> 🐫 case

Reply:
The rest I agree with you on. 

Unnnnnfortunately...

There are dozens of downstream repositories that already use three `snake_case` pieces of the `tabulate` API in hundreds of unique lines:

* `tabulate_formats`
* `multiline_formates`
* `tabular_data`

So, this is my attempt to keep the downstream changes after this PR down to the bare minimum of `import` changes only.

Thoughts?

Reply:
Okay, based on your opinions and @jakehader , I will just make these API changes, and make everything `camelCase`. NBD.